3-0-spring-framework/3-3-0-enable-string-trimming/src/main/java/com/bobocode/TrimmedAnnotationBeanPostProcessor.java

```diff
@@ -1,21 +1,106 @@
 package com.bobocode;
 
 import com.bobocode.annotation.Trimmed;
+import java.lang.reflect.Method;
+import java.lang.reflect.Parameter;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.stream.Stream;
+import lombok.SneakyThrows;
+import org.springframework.beans.BeansException;
 import org.springframework.beans.factory.config.BeanPostProcessor;
+import org.springframework.cglib.proxy.Callback;
+import org.springframework.cglib.proxy.Enhancer;
+import org.springframework.cglib.proxy.MethodInterceptor;
 import org.springframework.context.annotation.Configuration;
 import org.springframework.stereotype.Component;
+import org.springframework.util.StringUtils;
 
 /**
  * This is processor class implements {@link BeanPostProcessor}, looks for a beans where method parameters are marked with
  * {@link Trimmed} annotation, creates proxy of them, overrides methods and trims all {@link String} arguments marked with
  * {@link Trimmed}. For example if there is a string " Java   " as an input parameter it has to be automatically trimmed to "Java"
  * if parameter is marked with {@link Trimmed} annotation.
  * <p>
- *
+ * <p>
  * Note! This bean is not marked as a {@link Component} to avoid automatic scanning, instead it should be created in
  * {@link StringTrimmingConfiguration} class which can be imported to a {@link Configuration} class by annotation
  * {@link EnableStringTrimming}
  */
-public class TrimmedAnnotationBeanPostProcessor {
-//todo: Implement TrimmedAnnotationBeanPostProcessor according to javadoc
+public class TrimmedAnnotationBeanPostProcessor implements BeanPostProcessor {
+
+    private final Map<String, Object> beansToBeProxied = new HashMap<>();
+    private final Map<String, MethodInterceptor[]> beanInterceptors = new HashMap<>();
+
+    @Override
+    public Object postProcessBeforeInitialization(Object bean, String beanName) throws BeansException {
+        if (containsParametersAnnotatedWithTrimmed(bean)) {
+            beansToBeProxied.put(beanName, bean);
+            beanInterceptors.put(beanName, new MethodInterceptor[]{provideTrimmingInterceptor()});
+        }
+        return BeanPostProcessor.super.postProcessBeforeInitialization(bean, beanName);
```

Comment:
@Stanislav-Zabramnyi you can go with `super.postProcessBeforeInitialization(bean, beanName);`

Reply:
sure, this is just a default IDEA overriding